src/index.ts

```diff
@@ -132,15 +134,59 @@ function addSegmentFromDirectionsResponse(previousLngLat: LngLat, e: MapMouseEve
     });
 }
 
-function addSegmentFromStraightLine(previousLngLat: LngLat, e: MapMouseEvent): void {
-  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, e.lngLat);
+function addSegmentFromStraightLine(previousLngLat: LngLat, lngLat: LngLat, animate = true): void {
+  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, lngLat);
 
-  animationService.animateSegment(newSegment);
-  const marker = addMarker(e.lngLat, false);
+  if (animate) animationService.animateSegment(newSegment);
+  const marker = addMarker(lngLat, false);
   currentRun.addSegment(newSegment, marker);
   updateLengthElement();
 }
 
+function runToJson(run: CurrentRun): string {
+  if (run === undefined) return "{}";
+  let runJSON: {[name:string]: any} = {
+    start: {
+      lng: run.start.lngLat.lng,
+      lat: run.start.lngLat.lat,
+    },
+    distance: run.distance,
+    segments: [],
+    followRoads
+  }
+  for (let i in run.segments) {
+    runJSON.segments.push({
+      lng: run.segments[i].lngLat.lng,
+      lat: run.segments[i].lngLat.lat,
+      followsRoads: run.segments[i].followsRoads
```

Comment:
I still need to pull this branch down locally to confirm, but is `i` a RunSegment where we could just declare this as
```
lng: i.lngLat.lng,
lat: i.lngLat.lat,
...
```

Reply:
i is a string. a for-of loop would make i be a `RunSegment`